presets/presetManager.py

```diff
@@ -0,0 +1,223 @@
+import os.path
+import logging
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class PresetManager(object):
+    MAX_DEPTH = 2
+
+    def __init__(self, paths, strict=False):
+        self.presets = {}
+        self.strict = strict
+
+        # normalize paramter type string -> array
+        if isinstance(paths, basestring):
+            paths = [paths]
+
+        self._loadPaths(paths)
+
+    def _loadPaths(self, paths, depth=0):
+        if depth > self.MAX_DEPTH:
+            return
+
+        for i in range(len(paths)):
+            path = paths[i]
```

Comment:
> you mean for path in paths, right? :)

We can change the cycle but the point is that I do not want to do side-effect on ```paths```. Will it  change something on performance side?